app/decorators/case_transition_decorator.rb

```diff
@@ -35,6 +35,14 @@ def event_name
     I18n.t(specific_key, default: I18n.t(default_key))
   end
 
+  def rejected_case_creation_event
+    # This method prevents the case history changing from "rejected case created" to "case created"
+    # when a Case state is changed from "rejected" to another state
+    if object.case.has_attribute?(:case_originally_rejected) && (object.event == "create" && object.case.case_originally_rejected == true)
```

Comment:
You don't need to check a boolean value like that, it will evaluate as true or false directly.
Parentheses can be useful for comprehension, but probably not needed in this case.
```suggestion
    if object.case.has_attribute?(:case_originally_rejected) && object.event == "create" && object.case.case_originally_rejected
```